.github/workflows/ci_cd_test.yml

```diff
@@ -5,108 +5,120 @@ name: PR build & test
-        badge_line=$(awk '/!\[pylint\]/ {print FNR}' README.md)
-        sed -i "${badge_line}s#.*#${badge}#" README.md
-
-    - name: Pushing changes if any
-      uses: stefanzweifel/git-auto-commit-action@v4
-      with:
-        commit_message: Updating pylint score (from Github Action)
-        commit_user_name: sbacchio
-        commit_user_email: s.bacchio@gmail.com
-        commit_author: Simone Bacchio <s.bacchio@gmail.com>
-
-    - name: Run tests
-      run: |
-        pytest -v
-
-    - name: Run lyncs_setuptools
-      run: |
-        lyncs_setuptools
-
-    - name: Upload if not up to date
-      env:
-        TWINE_USERNAME: __token__
-        TWINE_PASSWORD: ${{ secrets.test_pypi_password }}
-      run: |
-        pip uninstall -y lyncs_io
-        pip install --extra-index-url https://test.pypi.org/simple/ lyncs_io==$(lyncs_setuptools version) || (
-          pip install twine
-          python setup.py sdist
-          twine upload --repository-url https://test.pypi.org/legacy/ dist/*
-          count=0
-          # waiting for pip to update the index
-          sleep 60
-        )
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - name: Install MPI (${{ matrix.mpi }})
+        run: bash .bash/install-${{ matrix.mpi }}.sh
+
+      - name: Set up Python ${{ matrix.py-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.py-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+
+      - name: Applying black formatting
+        run: |
+          pip install black
+          black --diff .
+          black .
+
+      - name: Pushing changes if any
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Applying black formatting (from Github Action)
+          commit_user_name: sbacchio
+          commit_user_email: s.bacchio@gmail.com
+          commit_author: Simone Bacchio <s.bacchio@gmail.com>
+
+      - name: Install from source
+        run: |
+          pip install -e .[all]
+
+      - name: Pylint output
+        run: |
+          pip install lyncs_setuptools[pylint]
+          badge=$(lyncs_pylint_badge --fail-under 8 . | sed "s/\&/\\\&/g")
+          badge_line=$(awk '/!\[pylint\]/ {print FNR}' README.md)
+          sed -i "${badge_line}s#.*#${badge}#" README.md
+
+      - name: Pushing changes if any
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Updating pylint score (from Github Action)
+          commit_user_name: sbacchio
+          commit_user_email: s.bacchio@gmail.com
+          commit_author: Simone Bacchio <s.bacchio@gmail.com>
+
+      - name: Run tests
+        run: |
+          pytest -v
+
+      - name: Run lyncs_setuptools
+        run: |
+          lyncs_setuptools
+
+      - name: Upload if not up to date
```

Comment:
I was planning to change This workflow drastically.
If you see now I have split the part of black & pylint in one, and test in the other.
The additional part of this is only the deployment on https://test.pypi.org and test of its installantio via pip.
So do not worry on keeping it up to date. I will fix it later!